packages/api/src/external/fhir/shared/hydrate-bundle.ts

```diff
@@ -38,7 +38,12 @@ function ensureUuidsAndAddExtensions(bundle: ValidBundle, docExtension: Extensio
         new: uuidv7(),
       });
     }
-    entry.resource.extension = [metriportDataSourceExtension, docExtension];
+    if (entry.resource.extension) {
+      entry.resource.extension.push(metriportDataSourceExtension);
+      entry.resource.extension.push(docExtension);
+    } else {
+      entry.resource.extension = [metriportDataSourceExtension, docExtension];
+    }
   });
   let bundleString = JSON.stringify(bundle);
   replacements.forEach((idPair: ReplacementIdPair) => {
```

Comment:
one lingering issue. If the oldId is undefined because it doesnt exist, this replace wont work right?

Reply:
if the oldid is undefined though, nothing can reference it so maybe this is okay then?

Reply:
yea, exactly

Reply:
no wait except the resource that didnt have a uuid will neveer get the new uuid and so it wont be valid when we try to insert into fhir server

Reply:
omg yes, good catch